### ISSUE TITLE
The prerprocess of input image for the Unet in training process

### ISSUE THREAD

**Issue by liupengcnu** [levihsu/OOTDiffusion#119]
Hi, Dr Xu, the following is my data preprocessing and training code when training ootdiffusion. Please check where there is any problem. 

            image_garm = vae_image_processor.preprocess(batch["cloth_values"])
            image_vton = vae_image_processor.preprocess(batch["masked_img_values"])
            image_ori = vae_image_processor.preprocess(batch["image_values"])

            garm_latents = vae.encode(image_garm).latent_dist.mode()
            vton_latents = vae.encode(image_vton).latent_dist.mode()
            image_ori_latents = vae.encode(image_ori).latent_dist.mode()

            image_ori_latents = image_ori_latents * vae.config.scaling_factor

            if random.uniform(0, 1) < 0.1:
                garm_latents = torch.zeros(garm_latents.shape)

            noise = torch.randn_like(image_ori_latents)
            noisy_latents = noise + image_ori_latents

            _, spatial_attn_outputs = unet_garm(
                garm_latents,
                0,
                encoder_hidden_states=prompt_embeds,
            )

            latent_vton_model_input = torch.cat([noisy_latents, vton_latents], dim=1)
            noise_pred = unet_vton(
                latent_vton_model_input,
                spatial_attn_outputs,
                timesteps,
                encoder_hidden_states=prompt_embeds,
            ).sample

            loss = F.mse_loss(noise_pred.float(), target.float(), reduction="mean")

**Comment by liupengcnu** [comment]
@levihsu Could you please help me point out the error in the above code?

**Comment by liupengcnu** [comment]
@levihsu  Hi, Dr Xu, could you give me some guidance for the above code?

**Comment by liupengcnu** [comment]
@T-Gu @ShineChen1024 Could you give me some guidance for the above code?

**Comment by levihsu** [comment]
@liupengcnu Hi. We really appreciate your enthusiasm in this project. But we don't have time to answer so many too detailed questions of yours one by one. We will release our training code in the future. Sorry for that and thanks for your understanding : )
